File: app/src/main/java/com/ladajules/notflix/utils/PreferenceManager.java
package com.ladajules.notflix.utils;

import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceManager {

    private final SharedPreferences sharedPreferences;

    public PreferenceManager(Context context) {
        this.sharedPreferences = context.getSharedPreferences(Constants.PREFERENCES_NAME, Context.MODE_PRIVATE);
    }

    public boolean isLoggedIn() {
        return sharedPreferences.getBoolean(Constants.KEY_IS_LOGGED_IN, false);
    }

    public void setLoggedIn(boolean value) {
        sharedPreferences.edit().putBoolean(Constants.KEY_IS_LOGGED_IN, value).apply();
    }

    public String getUserId() {
        return sharedPreferences.getString(Constants.KEY_USER_ID, null);
    }

    public void setUserId(String value) {
        sharedPreferences.edit().putString(Constants.KEY_USER_ID, value).apply();
    }

    public boolean isRememberMe() {
        return sharedPreferences.getBoolean(Constants.KEY_REMEMBER_ME, false);
    }

    public void setRememberMe(boolean value) {
        sharedPreferences.edit().putBoolean(Constants.KEY_REMEMBER_ME, value).apply();
    }

    // Onboarding
    public boolean isOnboardingCompleted() {
        return sharedPreferences.getBoolean(Constants.KEY_ONBOARDING_COMPLETED, false);
    }

    public void setOnboardingCompleted(boolean value) {
        sharedPreferences.edit().putBoolean(Constants.KEY_ONBOARDING_COMPLETED, value).apply();
    }

    // Profile
    public String getSelectedProfileId() {
        return sharedPreferences.getString(Constants.KEY_SELECTED_PROFILE_ID, null);
    }

    public void setSelectedProfileId(String value) {
        sharedPreferences.edit().putString(Constants.KEY_SELECTED_PROFILE_ID, value).apply();
    }

    public void clearAll() {
        sharedPreferences.edit().clear().apply();
    }

    // clear user session but keep onboarding status
    public void clearSession() {
        setLoggedIn(false);
        setUserId(null);
        setSelectedProfileId(null);
        setRememberMe(false);
    }

}

File: app/src/main/java/com/ladajules/notflix/utils/PreferenceManager.kt
package com.ladajules.notflix.utils

import android.content.Context
import android.content.SharedPreferences

class PreferenceManager(context: Context) {

    private val sharedPreferences: SharedPreferences =
        context.getSharedPreferences(Constants.PREFERENCES_NAME, Context.MODE_PRIVATE)

    // User Authentication
    var isLoggedIn: Boolean
        get() = sharedPreferences.getBoolean(Constants.KEY_IS_LOGGED_IN, false)
        set(value) = sharedPreferences.edit().putBoolean(Constants.KEY_IS_LOGGED_IN, value).apply()

    var userId: String?
        get() = sharedPreferences.getString(Constants.KEY_USER_ID, null)
        set(value) = sharedPreferences.edit().putString(Constants.KEY_USER_ID, value).apply()

    var rememberMe: Boolean
        get() = sharedPreferences.getBoolean(Constants.KEY_REMEMBER_ME, false)
        set(value) = sharedPreferences.edit().putBoolean(Constants.KEY_REMEMBER_ME, value).apply()

    // Onboarding
    var onboardingCompleted: Boolean
        get() = sharedPreferences.getBoolean(Constants.KEY_ONBOARDING_COMPLETED, false)
        set(value) = sharedPreferences.edit().putBoolean(Constants.KEY_ONBOARDING_COMPLETED, value).apply()

    // Profile
    var selectedProfileId: String?
        get() = sharedPreferences.getString(Constants.KEY_SELECTED_PROFILE_ID, null)
        set(value) = sharedPreferences.edit().putString(Constants.KEY_SELECTED_PROFILE_ID, value).apply()

    // Clear all preferences on logout
    fun clearAll() {
        sharedPreferences.edit().clear().apply()
    }

    // Clear user session but keep onboarding status
    fun clearSession() {
        isLoggedIn = false
        userId = null
        selectedProfileId = null
        rememberMe = false
    }
}